web/blog/tenzir-for-splunk-users/index.md

```diff
@@ -0,0 +1,316 @@
+
+## Why not SQL?
+
+Splunk was the first tool that provided an integrated solution from interactive
+data exploration to management-grade dashboards-all powered by dataflow
+pipelines. The success of Splunk is not only resulting from their first-mover
+advantage in the market, but also because their UX truly resonated with users:
+their target audience *liked* the user experience. It was *easy* to get things
+done.
+
+At Tenzir, we have a very clear target audience: security practitioners. They
+are not necessarily data engineers and fluent in SQL and low-level data tools,
+but rather identify as blue teamers, incident responders, threat hunters,
+detection engineers, threat intelligence analysts, and other domain experts. Our
+goal is cater to these folks, without requiring them to have deep understanding
+of relational algebra.
+
+The reason why we opted for a dataflow language is that it simplifies the
+reasoning: one step at a time. At least conceptually, because a smart system
+optimizes the execution of the users intent so that the pipeline runs
+efficiently. Any declarative language has that property, SQL in particular. But
+the difference to SQL is that a sequential dataflow makes it easy to simply
+*append* your next operation, as opposed to editing a query in the middle to
+inject a complicated clause somewhere. Instead of having to understand the
+entire SQL query, you just need to append an operation at the end that does the
+next thing. This dataflow pipeline style is becoming more an more popular. Most
+SIEMs have a language of their own, like Splunk. [Kusto][kusto] is another great
+example with a wide user base in security.
+
+[kusto]: https://learn.microsoft.com/en-us/azure/data-explorer/kusto/query/
+
+In fact, there's often an isomorphic SQL expression because a lot of the
+underlying engines internally map to the same Volcano-style model of execution.
+This gives rise to [transpiling dataflow languages to other execution
+platforms][splunk-transpiler]. Ultimately, our goal is that security
+practitioners do not have to think *any* of this and stay in their happy place,
+which means avoiding context switches to lower-level data primitives.
+
+[splunk-transpiler]: https://www.databricks.com/blog/2022/12/16/accelerating-siem-migrations-spl-pyspark-transpiler.html
+
+Now that we got the SQL topic out of the way, let's dive into some hands-on
+examples that illustrate the similarities and differences of SPL and TQL.
+
+## Examples
+
+Back in 2020, Eric Ooi wrote about [threat hunting with
+Zeek](https://www.ericooi.com/zeekurity-zen-part-iv-threat-hunting-with-zeek/),
+providing a set of Splunk queries that are corner stones for threat hunting.
+
+### Connections to destination port > 1024
+
+Splunk:
+
+```splunk-spl
+index=zeek sourcetype=zeek_conn id.resp_p > 1024
+| chart count over service by id.resp_p
+```
+
+Tenzir:
+
+```kusto
+export
+| where #schema == "zeek.conn" && id.resp_p > 1024
+| summarize count(.) by id.resp_p
```

Comment:
@jachris do think this is feasible until Thu or should we table it for later?

Reply:
I think we should introduce a small hack to make this possible, as it is a common pattern and otherwise quite awkward to express. I'm working on it right now.